pkg/project/registry/project/proxy/proxy.go

```diff
@@ -167,6 +182,34 @@ func (s *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectI
 	return projectutil.ConvertNamespace(namespace), false, nil
 }
 
+func (s *REST) tryImpersonatingNamespaceUpdate(project *api.Project, ctx kapi.Context) (*kapi.Namespace, bool) {
+	user, ok := kapi.UserFrom(ctx)
+	if !ok {
+		return nil, false
+	}
+
+	impersonatingClient, err := authclient.NewImpersonatingKubernetesClientset(user, s.config)
```

Comment:
The audit flow as I see it now:

**Success case**

1. Request by user to update project
2. No validation error
3. Request by loopback client to update namespace
4. Success

**Failure case**

1. Request by user to update project
2. Validation error
3. Failure

---

The audit flow if this PR is merged:

**Success case 1**

1. Request by user to update project
2. No validation error
3. Request by loopback client to update namespace
4. Success

**Success case 2**

1. Request by user to update project
2. Immutable field error
3. Request by loopback client that is impersonating the user to update namespace
4. User can update namespaces
5. Success

**Failure case 1**

1. Request by user to update project
2. Non immutable field error
3. Failure

**Failure case 2**

1. Request by user to update project
2. Immutable field error
3. Request by loopback client that is impersonating the user to update namespace
4. User cannot update namespaces
5. Failure